### PR TITLE
Improve Italian translation

### PR DESCRIPTION
**Pull request by matteocontrini**
**What does this PR change? What problem does it solve?**

Improve Italian translation and add missing translation keys.

**Was the change discussed in an issue or in the Discussions before?**

No.

## PR Checklist

- [x] This change adds/updates translations and I have used the [template present here](https://github.com/adityatelange/hugo-PaperMod/wiki/Translations#want-to-add-your-language-).
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have verified that the code works as described/as intended.
- [ ] This change adds a Social Icon which has a permissive license to use it.
- [x] This change **does not** include any CDN resources/links.
- [x] This change **does not** include any unrelated scripts such as bash and python scripts.
- [ ] This change updates the overridden internal templates from HUGO's repository.

### CODE REVIEW

**Review by adityatelange — COMMENTED** [review]
@matteocontrini can we include changes stated in below issue as well?
- #914

**Comment by matteocontrini** [comment]
Done.

**Comment by adityatelange** [comment]
> Done.

Thank you

File: i18n/it.yaml [hunk 1]
```diff
@@ -15,11 +15,17 @@
     other: "{{ .Count }} parole"
 
 - id: toc
-  translation: "Tabella dei Contenuti"
```
Comment (adityatelange) [inline i18n/it.yaml:18]: `Tabella dei Contenuti` fits more for Table of content a/c to google translate, let us keep this as it is

**Review by adityatelange — APPROVED** [review]

File: i18n/it.yaml (hunk 1 above)
Comment (matteocontrini) [reply i18n/it.yaml]: > `Tabella dei Contenuti` fits more for Table of content a/c to google translate, let us keep this as it is

I'm sorry, but I'm Italian native and I'm pretty sure I know Italian better than Google Translate... I've never seen `Tabella dei contenuti` used in in anything serious, it's a literal translation of `Table of contents` and doesn't really mean much in Italian, it's not idiomatic.

Wikipedia itself calls it `Indice dei contenuti`. Also, Google Translate offers two translations for me: `Sommario` and `Indice dei contenuti`, not `Tabella dei contenuti`. DeepL says `Indice dei contenuti`. Finally, in Italian we never use capital letters for common nouns, like in `Contenuti`, it's considered an error and there are no exceptions.

Comment (adityatelange) [reply i18n/it.yaml]: sure, understood, will push this change, my bad

Comment (matteocontrini) [reply i18n/it.yaml]: Thanks! 😃

Comment (adityatelange) [reply i18n/it.yaml]: Done https://github.com/adityatelange/hugo-PaperMod/commit/d67462d0f3f22cc5687f73f1edd454aa9cfa3800

**Pull request merged**